omegatrade-app/backend/app/controller/company.controller.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+const Company = require('../models/company.model')
+const { v4: uuidv4 } = require('uuid');
+
+/**
+ * Function to list all companies
+ * @method GET
+ */
+exports.getList = async function (req, res) {
+    try {
+        await Company.getAll(function (err, data) {
+            if (err)
+                return res.json({ success: false, message: "Error occured while fetching all Company" });
+            if (data) {
+                return res.status(200).json({ success: true, data: data });
+            }
+        });
+    } catch (error) {
+        return res.status(500).json({ success: false, message: error });
+    }
+};
+
+/**
+ * Function to create company
+ * @method POST
+ * @param {JSON} req contains request headers and payload companyName,companyShortCode,companyId,created_at
+ */
+exports.create = async function (req, res) {
+    try {
+        const body = req.body;
+        const company = await Company.checkCompany(body.companyName, body.companyShortCode);
+        if (company && company.length > 0) {
+            return res.status(409).json({ success: false, message: "company already exists!" });
+        } else {
+            body.companyId = uuidv4();
+            body.created_at = 'spanner.commit_timestamp()';
+            await Company.create(body, function (err, data) {
+                if (err) {
+                    return res.json({ success: false, message: "Error occured while creating company" });
+                }
+                if (data) {
+                    return res.status(200).json({ success: true, message: "company created successfully" });
+                }
+            });
+        }
+    } catch (error) {
+        return res.status(500).json({ success: false, message: "Company Insertion Failed!" });
```

Comment:
In the fetch function above, an error returns the actual error message. Here it returns a general error message. I think that a general error message is better when an unexpected error occurs, as it reveals less about the system and possible attack angles, but returning the actual error message could also be valid choice. I would however recommend that these functions are consistent with each other, and either all return a generic error message or the actual error message.

Reply:
changed to genric message in all functions.